brian2/equations/equations.py

```diff
@@ -426,6 +426,18 @@ def __init__(self, type, varname, dimensions, var_type=FLOAT, expr=None,
         # will be set later in the sort_subexpressions method of Equations
         self.update_order = -1
 
+        #: Mark the list of attributes that should not be considered for caching
+        #: of state update code, etc.
+        self._cache_irrelevant_attributes = ['update_order']
+
+    @property
+    def _state_tuple(self):
+        '''A tuple representing the full state of this object, used for
+        hashing and equality testing.'''
+        return tuple(value for key, value in self.__dict__.iteritems()
+                     if key not in (self._cache_irrelevant_attributes +
+                                    ['_cache_irrelevant_attributes']))
```

Comment:
I considered it, but given that the function is almost trivial, I'm not sure it's worth introducing a common super-class for `SingleEquation` and `Variable` given that the classes don't have much in common... Hmm, maybe something like `CacheKey` could make sense, I'll have a look.